CRM/Utils/System/Drupal.php

```diff
@@ -313,9 +313,10 @@ public function authenticate($name, $password, $loadCMSBootstrap = FALSE, $realP
 
     $config = CRM_Core_Config::singleton();
 
-    $ufDSN = CRM_Utils_SQL::autoSwitchDSN($config->userFrameworkDSN);
+    $ufDSN = $config->userFrameworkDSN;
```

Comment:
N.B. This variable (`$ufDSN`) is only used for two things: `DB::connect($ufDSN)` and `new Exception("Cannot connect...$ufDSN...")`.

The error message may look *slightly* different, but it's not a meaningful difference. 

What is meaningful is that `DB::connect()` is replaced by `CRM_Utils_SQL::connect()`, which behaves more reliably.